### PR TITLE
Support ActionMenu having null children

### PR DESCRIPTION
**Pull request by y-lohse**
In Drive we have an ActionMenu but not all items are visible at all time — if one of the items should be hidden, it returns null instead, as is common in React. This was breaking `ActionMenu` because it was checking `null.type`.

I also added an example for the `right` prop of ActionMenuItem so that we have visual regression tests for it.

### CODE REVIEW

**Comment by ptbrowne** [comment]
Can you add a test for the null case ?

**Comment by Crash--** [comment]
> In Drive we have an ActionMenu but not all items are visible at all time — if one of the items should be hidden, it returns null instead, as is common in React 

We're thinking about that and it's not impossible that instead of hiding features, we'll simply "disabled them" by displaying them with a special style and removing the onclick action.

**Comment by y-lohse** [comment]
> We're thinking about that and it's not impossible that instead of hiding features, we'll simply "disabled them" by displaying them with a special style and removing the onclick action.

Some will likely stay fully hidden, for example those that we want either in the topbar or in the menu depending on the context.

> Looking good, but can you add a test for the null case ?

Yup, done !

File: react/ActionMenu/index.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  afterAll(() => {
+    global.document.createRange = createRangeBackup
+  })
+
+  it('should support null children', () => {
+    const comp = mount(
+      <ActionMenu onClose={jest.fn()}>
+        <ActionMenuItem>Item 1</ActionMenuItem>
+        null
```
Comment (ptbrowne) [inline react/ActionMenu/index.spec.jsx:28]: This is a null string here ?

I think you want `{ null }`.

File: react/ActionMenu/index.spec.jsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  beforeAll(() => {
+    createRangeBackup = global.document.createRange
+    global.document.createRange = jest.fn(() => ({
+      setStart: () => {},
+      setEnd: () => {},
+      commonAncestorContainer: {
+        nodeName: 'BODY',
+        ownerDocument: document
+      }
+    }))
```
Comment (ptbrowne) [inline react/ActionMenu/index.spec.jsx:17]: Why is this needed ?

Comment (Crash--) [reply react/ActionMenu/index.spec.jsx:17]: We have issue for testing component using Tooltip from MUI. See https://github.com/mui-org/material-ui/issues/15726#issuecomment-493124813 

jsdom-16 fix the issue, but we need to use https://www.npmjs.com/package/jest-environment-jsdom-sixteen. I tried few months ago but still the issue. Maybe it's fine now.

File: react/ActionMenu/index.spec.jsx (hunk 1 above)
Comment (y-lohse) [reply react/ActionMenu/index.spec.jsx:28]: 🤦‍♂️thanks

File: react/ActionMenu/index.spec.jsx (hunk 2 above)
Comment (ptbrowne) [reply react/ActionMenu/index.spec.jsx:17]: AMHA this should not be in this test file and should be called with a name that reviewers can understand.

Tooltip-testing.js

```
const fixTooltipTesting = () => {
  let createRangeBackup
  beforeAll(() => {
    createRangeBackup = global.document.createRange
    global.document.createRange = jest.fn(() => ({
      setStart: () => {},
      setEnd: () => {},
      commonAncestorContainer: {
        nodeName: 'BODY',
        ownerDocument: document
      }
    }))

  afterAll(() => {
    global.document.createRange = createRangeBackup
  })
}
```

This way we do not need a comment, it's clearer for the reviewer.

Comment (Crash--) [reply react/ActionMenu/index.spec.jsx:17]: and we can import it from other libs 👍

File: react/ActionMenu/index.spec.jsx (hunk 1 above)
Comment (ptbrowne) [reply react/ActionMenu/index.spec.jsx:28]: Was the testing passing with the "null" string ? If yes, this could be a problem.

File: react/Popper/testing.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      }
+    }))
+  })
+
+  afterAll(() => {
+    global.document.createRange = createRangeBackup
+  })
+}
+
+export default fixPopperTesting
```
Comment (ptbrowne) [inline react/Popper/testing.js:23]: Nit: since this file could have other functions (helping to interact with a Popper from the test), I think it would be better to have a normal export instead of a default one.

As a user, I would try `import { fixPopperTesting } from './testing'` since the file is named `testing` and not `fixPopperTesting.js`

**Comment by ptbrowne** [comment]
👍
